src/core/api.ts

```diff
@@ -6,15 +6,22 @@ import { GET_CHROM_SIZES } from './utils/assembly';
 import { CompleteThemeDeep } from './utils/theme';
 import { traverseViewsInViewConfig } from './utils/view-config';
 
-export type CommonEventData = {
-    genomicPosition: string;
-    data: Datum[];
-};
-
-export type RawDataEventData = {
+export interface CommonEventData {
+    /** Source visualization ID, i.e., `track.id` */
     id: string;
+    /** Values in a JSON array that represent data after data transformation */
     data: Datum[];
-};
+}
+
+export interface PointMouseEventData extends CommonEventData {
+    /* A genomic coordinate, e.g., `chr1:100,000`. */
+    genomicPosition: string;
```

Comment:
Any reason this isn't `[name: string, position: number]`? It's somewhat trival to convert to string representation by end user but not the other way around.

Reply:
Will address this together with the next comment.